src/kOS/AddOns/KerbalAlarmClock/KACAlarmWrapper.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Linq;
+using kOS.Safe.Encapsulation;
+using kOS.Safe.Encapsulation.Suffixes;
+using kOS.Safe.Utilities;
+using kOS.AddOns.KerbalAlarmClock;
+
+namespace kOS.Suffixed
+{
+    public class KACAlarmWrapper : Structure
+    {
+        private KACWrapper.KACAPI.KACAlarm alarm;
+        private SharedObjects shared;
+
+        public KACAlarmWrapper(KACWrapper.KACAPI.KACAlarm init, SharedObjects shared) 
+        { 
+            alarm = init;
+            this.shared = shared;
+            InitializeSuffixes();
+        }
+        public KACAlarmWrapper(String alarmID, SharedObjects shared) 
+        {
+            alarm = KACWrapper.KAC.Alarms.First(z=>z.ID==alarmID);
+            this.shared = shared;
+            InitializeSuffixes();
+        }
+        private void InitializeSuffixes()
+        {
+            AddSuffix("ID", new Suffix<string>(() => alarm.ID));
+            AddSuffix("NAME", new SetSuffix<string>(() => alarm.Name, value => alarm.Name = value));
+
+            AddSuffix("VESSELID", new SetSuffix<string>(() => alarm.VesselID, value => alarm.VesselID = value));
+            AddSuffix("VESSEL", new SetSuffix<VesselTarget>(() => getVesselByID(alarm.VesselID), value => alarm.VesselID = value.Vessel.id.ToString()));
+
+            AddSuffix("NOTES", new SetSuffix<string>(() => alarm.Name, value => alarm.Name = value));
+
+            AddSuffix("ALARMACTION", new SetSuffix<string> (alarm.AlarmAction.ToString, SetAlarmAction));
+
+            AddSuffix("ALARMTYPE", new Suffix<string> (alarm.AlarmType.ToString));
+
+            AddSuffix("REMAINING", new Suffix<double>(GetRemaining));
+
+            AddSuffix("ALARMTIME", new SetSuffix<double>(() => alarm.AlarmTime, value => alarm.AlarmTime = value));
+            AddSuffix("ALARMMARGIN", new SetSuffix<double>(() => alarm.AlarmMargin, value => alarm.AlarmMargin = value));
+
+            AddSuffix("REPEATALARM", new SetSuffix<Boolean>(() => alarm.RepeatAlarm, value => alarm.RepeatAlarm = value));
+
+            AddSuffix("REPEATALARMPERIOD", new SetSuffix<double>(() => alarm.RepeatAlarmPeriod, value => alarm.RepeatAlarmPeriod = value));
+
+            AddSuffix("XferOriginBodyName", new SetSuffix<string>(() => alarm.XferOriginBodyName, value => alarm.XferOriginBodyName = value));
+            AddSuffix("XferTargetBodyName", new SetSuffix<string>(() => alarm.XferTargetBodyName, value => alarm.XferTargetBodyName = value));

```

Comment:
these could simply be "ORIGINBODY" and "TARGETBODY". because we already have a BODYTARGET type. we could give them the body itself, making this suffix much shorter. 

if they are here only for transfer alarms we might consider only having these suffixes when they are appropriate